aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3ExpressDecorator.kt

```diff
@@ -183,6 +185,33 @@ class S3ExpressIdentityProviderConfig(codegenContext: ClientCodegenContext) : Co
         }
 }
 
+class S3ExpressFluentClientCustomization(
+    codegenContext: ClientCodegenContext,
+) : FluentClientCustomization() {
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val codegenScope =
+        arrayOf(
+            *preludeScope,
+            "S3ExpressRuntimePlugin" to s3ExpressModule(runtimeConfig).resolve("runtime_plugin::S3ExpressRuntimePlugin"),
+        )
+
+    override fun section(section: FluentClientSection): Writable =
+        writable {
+            when (section) {
+                is FluentClientSection.AdditionalBaseClientPlugins -> {
+                    rustTemplate(
+                        """
+                        .with_client_plugin(#{S3ExpressRuntimePlugin}::new(${section.config}.config.clone()))
+                        """,
+                        *codegenScope,
+                    )
```

Comment:
I would recommend changing the API of this section to be statements instead of chaining `.` calls onto the existing plugin builder. This is going to limiting (e.g. you can't call a method that accepts `&mut self`.

Alternatively, you could change the API to instead _solely_ contian a single client plugin and call `.with_client_plugin` at the root.